LogMonitor/src/LogMonitor/PipeMonitor.cpp

```diff
@@ -16,13 +15,39 @@ DWORD HandlePipeStream(HANDLE hPipe) {
     char chBuf[BUFSIZE];
     bool bSuccess = false;
 
+    // Loop until done reading
     while (1) {
-        bSuccess = ReadFile(hPipe, chBuf, BUFSIZE * sizeof(char), &dwRead, NULL);
+        // Read client requests from the pipe. only allows messages up to BUFSIZE characters in length.
+        bSuccess = ReadFile(
+            hPipe,      // handle to pipe 
+            chBuf,      // buffer to receive data
+            BUFSIZE * sizeof(char), // size of buffer
+            &dwRead,                // number of bytes read
+            NULL);                  // not overlapped I/O 
 
         DWORD readFileError = GetLastError();
         if (!bSuccess || dwRead == 0)
         {
-            logWriter.WriteConsoleLog(std::to_wstring(readFileError));
+            switch (readFileError)
+            {
+            case ERROR_BROKEN_PIPE:
+                logWriter.TraceError(
+                    Utility::FormatString(L"Client disconnected. Error: %lu", readFileError).c_str()
+                );
+                break;
+            case ERROR_MORE_DATA:

```

Comment:
What should we do in this case, read the message in bits then concatenate before writing to console?

Reply:
From the Microsoft docs they've handled this using a do while loop. We can borrow the approach in our scenario as well.


`
   do 
   { 
   // Read from the pipe. 
 
      fSuccess = ReadFile( 
         hPipe,    // pipe handle 
         chBuf,    // buffer to receive reply 
         BUFSIZE*sizeof(TCHAR),  // size of buffer 
         &cbRead,  // number of bytes read 
         NULL);    // not overlapped 
 
      if ( ! fSuccess && GetLastError() != ERROR_MORE_DATA )
         break; 
 
      _tprintf( TEXT("\"%s\"\n"), chBuf ); 
   } while ( ! fSuccess);  // repeat loop if ERROR_MORE_DATA 

`

Reply:
https://learn.microsoft.com/en-us/windows/win32/ipc/named-pipe-client?redirectedfrom=MSDN#:~:text=do%20%0A%20%20%20%7B%20%0A%20%20%20//%20Read,loop%20if%20ERROR_MORE_DATA

Reply:
Interesting, the approach to ignore ERROR_MORE_DATA seems to allow the text to be written correctly, I'll push the fix